File: common/script/menu.js
/*
 * Copyright © 2015 Matt Robinson
 *
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

(function () {
    "use strict";

    function showhide(hash)
    {
        $('.menu').hide();

        if(hash)
        {
            $(hash).show();
        }
        else
        {
            $('#menu1').show();
        }
    }

    $(document).ready(function() {
        if(!("onhashchange" in window))
        {
            return;
        }

        $(window).bind('hashchange', function()
        {
            showhide(window.location.hash);
        });

        showhide(window.location.hash);
    });
}());

File: _plugins/EBS/menu.rb
# frozen_string_literal: true

# Copyright © 2015-2024 Matt Robinson
#
# SPDX-License-Identifier: GPL-3.0-or-later

module EBS
  class Menu < Liquid::Drop
    def initialize
      super
      @entries = []
    end

    attr_accessor :title, :id, :entries
  end
end
